From 38f1448f240c338003f595bd4cd99e47d0ab2501 Mon Sep 17 00:00:00 2001
Message-Id: <38f1448f240c338003f595bd4cd99e47d0ab2501@dist-git>
From: Michal Privoznik <mprivozn@redhat.com>
Date: Wed, 19 Nov 2014 18:59:14 +0100
Subject: [PATCH] qemu: Support OVMF on armv7l aarch64 guests

https://bugzilla.redhat.com/show_bug.cgi?id=1165782

Currently, we are whitelisting architectures, that we know how to run
OVMF on. So far, only x86_64 was enabled. However, looking at qemu
code, the same commandline can be used to enable OVMF for armv7l and
aarch64.

Signed-off-by: Michal Privoznik <mprivozn@redhat.com>
(cherry picked from commit 6d8054b68407a3385b33c867a425ad8278b0b8f0)
Signed-off-by: Jiri Denemark <jdenemar@redhat.com>
---
 src/qemu/qemu_command.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/qemu/qemu_command.c b/src/qemu/qemu_command.c
index 6c4a8b5..2b8eae0 100644
--- a/src/qemu/qemu_command.c
+++ b/src/qemu/qemu_command.c
@@ -7643,7 +7643,9 @@ qemuBuildDomainLoaderCommandLine(virCommandPtr cmd,
 
     case VIR_DOMAIN_LOADER_TYPE_PFLASH:
         /* UEFI is supported only for x86_64 currently */
-        if (def->os.arch != VIR_ARCH_X86_64) {
+        if (def->os.arch != VIR_ARCH_X86_64 &&
+            def->os.arch != VIR_ARCH_ARMV7L &&
+            def->os.arch != VIR_ARCH_AARCH64) {
             virReportError(VIR_ERR_CONFIG_UNSUPPORTED,
                            _("pflash is not supported for %s guest architecture"),
                            virArchToString(def->os.arch));
-- 
2.1.3

